Frontend/src/CustomScripts/convertData.mjs

```diff
@@ -58,37 +74,53 @@ for (const key of Object.keys(SC1DataFormat)) {
     max,
   };
 
-  if (categoryIdx === -1) {
-    // doesn't exist = create new category
-    categories.push({
-      category,
-      values: [listing],
-    });
+  let res = category.split(';')
+
+  let c = res[0]
+  let subc = res[1]
+  let ind = undefined;
+  for (let j = 0; j < newCategories.length; j++) {
+    if (newCategories[j].category.localeCompare(c) == 0) {
+      ind = j
+      break
+    }
+  }
+
+  if (ind) {
```

Comment:
```suggestion
  // find the category that this goes into
  const ind = categories.findIndex((cat) => cat.category === c);

  if (ind !== -1) {
```
Rather than using a for loop to find the index of the category, you can make it more succinct by using `findIndex()` like this (also shown on lines 46-47 of the previous version of the diff).

Reply:
fixed